src/robot/output/listeners.py

```diff
@@ -74,6 +74,10 @@ def _import_listeners(self, listener_data):
                              % (listener, unicode(err)))
         return listeners
 
+    def imported(self, name, import_type):
+        for listener in self._listeners:
+            listener.call_method(listener.imported, name, import_type)
```

Comment:
When talking about attrs as additional parameter. Should it look like def imported(self, import_type, name, attrs) or shall I just pass the whole suite and attrs?
grumps
I've tried such approach before, but YAGNId it. Take a look: https://github.com/stasiek/robotframework/blob/import-validation/src/robot/output/importvalidator.py
grumps
There we'd have source, the rest of stuff and in suite.resource.variables - variables so we'll be able to proceed with import validation.